contracts/FTokenStrat.sol

```diff
@@ -0,0 +1,143 @@
+//SPDX-License-Identifier: Unlicense
+/*
+harvest finance strategy
+prox:         0xab7fa2b2985bccfc13c6d86b1d5a17486ab1e04c
+current_imp:  0x9b3be0cc5dd26fd0254088d03d8206792715588b
+fDAI:         0xe85c8581e60d7cd32bbfd86303d2a4fa6a951dac
+
+*/
+
+pragma solidity 0.7.3;
+
+import "./IStrat.sol";
+import "./IFToken.sol";
+import "./IVault.sol";
+import "./Timelock.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+
+contract FTokenStrat is IStrat {
+
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    using SafeERC20 for IERC20Detailed;
+    IVault public vault;
+    IFToken public fToken;
+    IERC20Detailed public underlying;
+    Timelock public timelock;
+    uint public immutable minWithdrawalCap; // prevents the owner from completely blocking withdrawals
+    uint public withdrawalCap = uint(-1); // max uint
+    uint public buffer; // buffer of underlying to keep in the strat
+    string public name = "Harvest Finance"; // for display purposes only
+
+    modifier onlyVault {
+        require(msg.sender == address(vault));
+        _;
+    }
+
+    modifier onlyOwner {
+        require(msg.sender == vault.owner()); // vault owner is strat owner
+        _;
+    }
+
+    constructor(IVault vault_, IFToken fToken_) {
+        vault = vault_;
+        fToken = fToken_;
+        timelock = Timelock(vault.timelock()); // use the same timelock from the vault
+        underlying = IERC20Detailed(fToken_.underlying());
+        underlying.safeApprove(address(fToken), uint(-1)); // intentional underflow
```

Comment:
"Intentional underflow" sounds like a disaster waiting to happen

Reply:
It's a common practice sometimes, essentially means a max-approve of the vault controlling the strategy.